sources/packages/backend/apps/api/src/route-controllers/application/_tests_/application.students.getCompletedApplicationDetails.e2e-spec.ts

```diff
@@ -0,0 +1,233 @@
+    const endpoint = `/students/application/${application.id}/completed`;
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect({
+        firstDisbursement: {
+          coeStatus: firstDisbursement.coeStatus,
+          disbursementScheduleStatus:
+            firstDisbursement.disbursementScheduleStatus,
+        },
+        secondDisbursement: {
+          coeStatus: secondDisbursement.coeStatus,
+          disbursementScheduleStatus:
+            secondDisbursement.disbursementScheduleStatus,
+        },
+        assessmentTriggerType: application.currentAssessment.triggerType,
+      });
+  });
+
+  it(`Should get application details with scholastic standing change type when application has a scholastic standing '${StudentScholasticStandingChangeType.StudentDidNotCompleteProgram} associated with.`, async () => {
+    // Arrange
+    const application = await saveFakeApplicationDisbursements(appDataSource, {
+      student,
+    });
+    application.applicationStatus = ApplicationStatus.Completed;
+    await applicationRepo.save(application);
+    const [firstDisbursement] =
+      application.currentAssessment.disbursementSchedules;
+    firstDisbursement.coeStatus = COEStatus.completed;
+    await disbursementScheduleRepo.save(firstDisbursement);
+    // Create a scholastic standing and have it associated with the completed application.
+    const scholasticStanding = createFakeStudentScholasticStanding({
+      submittedBy: submittedByInstitutionUser,
+    });
+    // 'Student did not complete program' is the only'scholastic standing that does not generate an assessment.
+    // The below record has only a relationship with the application which must be enough to
+    // have the scholasticStandingChangeType returned.
+    scholasticStanding.application = application;
+    await studentScholasticStandingRepo.save(scholasticStanding);
+
+    const endpoint = `/students/application/${application.id}/completed`;
+    const token = await getStudentToken(
+      FakeStudentUsersTypes.FakeStudentUserType1,
+    );
+    // Act/Assert
+    await request(app.getHttpServer())
+      .get(endpoint)
+      .auth(token, BEARER_AUTH_TYPE)
+      .expect(HttpStatus.OK)
+      .expect({
+        firstDisbursement: {
+          coeStatus: firstDisbursement.coeStatus,
+          disbursementScheduleStatus:
+            firstDisbursement.disbursementScheduleStatus,
+        },
+        assessmentTriggerType: application.currentAssessment.triggerType,
+        scholasticStandingChangeType:
+          StudentScholasticStandingChangeType.StudentDidNotCompleteProgram,
+      });
+  });
+
+  it(`Should get application details with the most updated appeal status when the application has more than one student appeals associated with.`, async () => {
```

Comment:
just to refresh my mind, when we say `most updated appeal status`.  is it the appeal that was created by the student most recently? or is it the last appeal that was accepted/declined by the ministry?

Reply:
got the answer with submitted date